Core/Core/Analytics/CoreAnalytics.swift

```diff
@@ -244,3 +254,41 @@ public struct EventCategory {
     public static let video = "video"
     public static let course = "course"
 }
+
+public enum EnrollmentMode: String {
+    case audit
+    case verified
+    case none
+}
+
+public enum Pacing: String {
+    case `self` = "self"
+    case instructor = "instructor"
+}
+
+public enum Action: String {
+    case on = "on"
+    case off = "off"
+    case allow = "Allow"
+    case doNotAllow = "donot_allow"
+    case ok = "OK"
+    case cancel = "Cancel"
+    case update = "Update"
+    case remove = "Remove"
+    case done = "Done"
+    case viewEvent = "view_event"
+}
+
+public enum Dialog: String {
+    case permission = "Permission"
+    case add = "Add"
+    case remove = "Remove"
+    case update = "Update"
+    case confirmed = "Confirmed"
+}
+
+public enum Snackbar: String {
+    case add = "Add"
+    case remove = "Remove"
+    case update = "Update"
```

Comment:
It would be nice if you could move this to Course Analytics and also keep all the actions, and stuff in snake case instead of camel case.